doc/sphinx/architecture.md

```diff
@@ -0,0 +1,252 @@
+# Overview
+This document describes the high-level architecture of *Mir*, including a diagram and a brief introduction of the libraries.
+
+# Audience
+This document is intended to provide contributors to *Mir* an overview of *Mir*'s systems. It is *not* intended to guide compositor-implementers, unless they are specifically interested in the internals of *Mir*.
+
+# Diagram
+![Architecture Diagram](./architecture.png "Architecture Diagram")
+
+# Top-Level Folders
+## `/include`
+- The `include` folder at the root of the project provides the public-facing API.
+- This directory *strictly* includes the interfaces that are useful to a compositor-implementer.
+- If one is only interested in implementing their own *Mir*-based compositor, they need not look any further than this directory.
+
+## `/src`
+- The `src` folder at the root of the project comprises the core implementation of *Mir*.
+
+## `/examples`
+- A collection of demo *Mir* compositors that are used as examples of how one might build a true compositor. Examples include:
+  - `miral-app`
+  - `miral-shel`
+  - `mir_demo_server`
+
+## `/tests`
+- Unit tests for the entire project.
+
+# Libraries
+The following is the library tree:
+```
+├── miral
+├── miroil
+├── mircore
+├── mircommon
+├── mircookie
+└── mirserver
+    ├── mircompositor
+    ├── mirinput
+    ├── mirfrontend
+    ├── mirfrontend-wayland
+    ├── mirfrontend-xwayland
+    ├── mirshell
+    ├── mirshelldecoration
+    ├── mirscene
+    ├── mirgraphics
+    ├── mirreport
+    ├── mirlttng
+    ├── mirlogging
+    ├── mirnullreport
+    ├── mirconsole
+    ├── mirgl
+    └── mirrenderergl
```

Comment:
_I_ wouldn't expand mirserver here. I think that introduces too many concepts.

Reply:
Sure, they need explaining but we're well past the number of concepts that should be introduced together. (Even Agatha Christie stops at a dozen suspects and she was aiming to confuse!)

First introduce [the high level concepts](https://github.com/MirServer/mir/pull/3053#pullrequestreview-1650985226). Then, break down each one in turn: (e.g. "These are the platforms... this one is for X11...", "This is the mirserver... it has these components...").

That gives the reader some structure to organise their understanding.